name: Spelling check for docs (Vale)
on:
  workflow_dispatch:
  push:
    branches:
      - main
      - "release/**"
    paths:
      - "docs/**"
  pull_request:
    paths:
      - "docs/**"

jobs:
  vale:
    runs-on: ubuntu-24.04
    steps:
      - name: Checkout
        uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # v4.2.2
        with:
          ref: ${{ !github.event.pull_request.head.repo.fork && github.head_ref || '' }}
      # Work around https://github.com/errata-ai/vale-action/issues/128.
      - run: |
          venv="$HOME/.local/share/venv"
          python3 -m venv "$venv"
          echo "$venv/bin" >> "$GITHUB_PATH"
      - name: Vale
        uses: errata-ai/vale-action@2690bc95f0ed3cb5220492575af09c51b04fbea9 # tag=reviewdog
        with:
          files: docs/docs
          fail_on_error: true
          version: 3.9.3
